app/livechat/server/lib/SNS.ts

```diff
@@ -8,15 +8,23 @@ const logger = new Logger('SNS');
 
 const addLogDataInSNSPayload = (roomId: string, data: Record<string, any>): Record<string, any> => {
 	const room = LivechatRooms.findOneLivechatById(roomId);
-	if (!room || !room.customFields?.logData) {
+	if (!room || !room.customFields?.sns?.logData) {
 		return data;
 	}
 
-	const { logData } = room.customFields;
+	const { logData } = room.customFields?.sns || {};
 	Object.assign(logData, data);
 	return logData;
 };
 
+const getSNSTopicARN = (roomId: string): string => {
+	const room = LivechatRooms.findOneLivechatById(roomId);
+	if (!room || !room.customFields?.sns) {
+		return '';
+	}
+	return room.customFields?.sns?.arn || '';
```

Comment:
```suggestion
	return room?.customFields?.sns?.arn || '';
```
Any reason not to simplify?

Reply:
Thanks Alexander!